### PR TITLE
Fix #13, add JoyFoodSunshine

### LINKED ISSUE #13: Cannot parse through first chocolate chip cookie recipe Google gave
[Here](https://joyfoodsunshine.com/the-most-amazing-chocolate-chip-cookies/) is the link to the unparseable recipe.

### PR DESCRIPTION
**Pull request by Eldemarkki**
Fixes #13

### CODE REVIEW

File: lib/domain/annoyingDomains.ts [hunk 1]
```diff
@@ -2,6 +2,25 @@ import { combineIngredientNamesAndAmounts, getItemListFromSelector, getTitleFrom
...
+            [key: string]: {
+                amount: string,
+                unit: string,
+                unit_id: number
+            }
+        },
+        unit_id: number,
+        id: number
+    }]
+}
```
Comment (Eldemarkki) [inline lib/domain/annoyingDomains.ts:22]: Is this needed, or is it fine to just use `any` in this case?

Comment (jpbulman) [reply lib/domain/annoyingDomains.ts:22]: Yeah I like the interface! Makes it safer/easier with the downstream inferencing

File: lib/domain/annoyingDomains.ts [hunk 2]
```diff
@@ -50,6 +69,44 @@ const getBonAppetitData = (html: string): RecipeMetadata => {
...
 
+const getJoyFoodSunshineData = (html: string): RecipeMetadata => {
+    const title = getTitleFromSelector(html, 'h2.wprm-recipe-name')
+
+    const $ = cheerio.load(html);
+
+    const directions: string[] = $("div.wprm-recipe-instruction-text").toArray().map(element => $(element).text().trim());
+
+    // The html contains the ingredients in a javascript array, so here we locate where that array is
+    const ingredientsStart = html.indexOf(' = {"ingredients":[');
```
Comment (jpbulman) [inline lib/domain/annoyingDomains.ts:80]: This is a great implementation that gets all of the details, nice! :fire:

**Comment by jpbulman** [comment]
I tried this with the recipe from the issue and another random carrot cake one, both seem to work fine to me! Thanks for the contribution, merging :slightly_smiling_face:

**Pull request merged**